changelog.d/20230622_141438_chris_endpoint_setup.rst

```diff
@@ -0,0 +1,17 @@
+.. A new scriv changelog fragment.
+..
+.. Uncomment the header that is right (remove the leading dots).
+..
+New Functionality
+^^^^^^^^^^^^^^^^^
+
+- Added ``endpoint_setup`` option to endpoint config, which, if present, is run by the
+  system shell during the endpoint startup process, before any other initialization.
```

Comment:
Seems to me that the previous conversation might be cleared up by linking to the appropriate place in the documentation.

Which leads me to wonder where the documentation for `endpoint_setup` is?

The logical place might be on the [Endpoints](https://funcx.readthedocs.io/en/latest/endpoints.html) page.